src/Form/ChoiceList/ModelChoiceLoader.php

```diff
@@ -128,7 +128,7 @@ public function loadChoiceList($value = null)
         if (!$this->choiceList) {
             if ($this->query) {
                 $entities = $this->modelManager->executeQuery($this->query);
-            } elseif (\is_array($this->choices) && \count($this->choices) > 0) {
+            } elseif (\is_array($this->choices)) {
```

Comment:
Given what is currently checked in the test, I think this is the only change required in this PR.

Reply:
Not only. If we keep ModelType with default option `choices`  with empty when on default entities dosent loaded

Reply:
Then, that isn't checked in the test. I mean, if I change only this line and run `simple-phpunit --filter ModelTypeTest::testGetOptions`, the test is passing.

Reply:
![image](https://user-images.githubusercontent.com/30481080/97623050-2cf8ef80-1a36-11eb-8e93-3f34513d46f4.png)
proofs

Reply:
I don't know how we could get different results.
This is the [codebase](https://github.com/sonata-project/SonataAdminBundle/compare/3.x...phansys:pr_6548?expand=1) I'm using in the test.
As you can see, without any modification in the property or the option, the test is passing:
```
vendor/bin/simple-phpunit --filter ModelTypeTest::testGetOptions -vvv --debug
PHPUnit 8.5.8 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.8
Configuration: /mnt/phpunit.xml.dist

Testing SonataAdminBundle Test Suite
Test 'Sonata\AdminBundle\Tests\Form\Type\ModelTypeTest::testGetOptions with data set #0 (null, 1)' started
Test 'Sonata\AdminBundle\Tests\Form\Type\ModelTypeTest::testGetOptions with data set #0 (null, 1)' ended
Test 'Sonata\AdminBundle\Tests\Form\Type\ModelTypeTest::testGetOptions with data set #1 (array(), 0)' started
Test 'Sonata\AdminBundle\Tests\Form\Type\ModelTypeTest::testGetOptions with data set #1 (array(), 0)' ended


Time: 513 ms, Memory: 28.00 MB

OK (2 tests, 32 assertions)
```

Reply:
I don't know too

Reply:
Im using not simple-phpunit

Reply:
> without any modification in the property or the option

Im using this branch(closes-issue-6548) and rollback changes in `ModelChoiceLoader`. And my test detect error.
Tests in master branch works incorrect - no check actually loaded entities.

If we remove check on counts of element we broke default behavior see ModelType `setDefaults('choices', [])`

Reply:
Becouse there needed chages too

Reply:
@phansys @VincentLanglet whyt?

Reply:
When you're passing explicitly null in the test, it's working indeed.
But what if you're not passing a choice.
```
$options = $optionResolver->resolve(['model_manager' => $modelManager]);
```
It should keep calling the `findBy`, but it won't anymore @phansys.

That's why I agree with @kirya-dev.

Reply:
I didn't analyzed the change, but if that's is fine, I think the test must cover the case (the test provided in this PR is passing even if these changes are not applied).

Reply:
@kirya-dev Can you add a test with `$options = $optionResolver->resolve(['model_manager' => $modelManager]);` ?

Reply:
Of course. Added!